storage/src/main/java/com/tencent/rss/storage/handler/impl/LocalFileClientReadHandler.java

```diff
@@ -53,29 +60,88 @@ public LocalFileClientReadHandler(
     this.shuffleServerClients = shuffleServerClients;
   }
 
-  @Override
-  public ShuffleDataResult readShuffleData(int segmentIndex) {
+  public ShuffleIndexResult readShuffleIndex() {
+    boolean readSuccessful = false;
+    ShuffleIndexResult shuffleIndexResult = null;
+    RssGetShuffleIndexRequest request = new RssGetShuffleIndexRequest(
+        appId, shuffleId, partitionId, partitionNumPerRange, partitionNum);
+
+    for (ShuffleServerClient shuffleServerClient : shuffleServerClients) {
+      try {
+        shuffleIndexResult = shuffleServerClient.getShuffleIndex(request).getShuffleIndexResult();
+        readSuccessful = true;
+        break;
+      } catch (Exception e) {
+        LOG.warn("Failed to read shuffle index with " + shuffleServerClient.getClientInfo(), e);
+      }
+    }
+
+    if (!readSuccessful) {
+      throw new RssException("Failed to read shuffle index for appId[" + appId + "], shuffleId["
```

Comment:
ShuffleServer also throw RssException? @colinmjj

Reply:
This handler is used in client

Reply:
No need to throw rss exception in server side now.